File: src/test/java/org/meetkt/catalogue/domain/model/ProductIdUnitTest.java
package org.meetkt.catalogue.domain.model;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertThrows;

class ProductIdUnitTest {

    @Test
    void shouldThrowExceptionGivenProductIdLessThanMinimumLength() {
        assertThrows(IllegalArgumentException.class, () -> new ProductId("P-1"));
    }
}

File: src/test/kotlin/in/meetkt/catalogue/domain/model/ProductIdUnitTest.kt
package `in`.meetkt.catalogue.domain.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ProductIdUnitTest {

    @Test
    fun shouldReturnTrueGiven2ProductIdsAreEqual() {
        val productId = ProductId("001")
        val otherProductId = ProductId("001")

        assertThat(productId).isEqualTo(otherProductId)
    }
}